src/helpers/chain.js

```diff
@@ -0,0 +1,582 @@
+ * @param {string} oracle address of the oracle
+ * @returns {object} an object containing feed invitation IDs. Each field in
+ *                   the object represents the feed name (Ex. ATOM-USD) and its
+ *                   value is a number which is the invitation ID.
+ */
+export const getOraclesInvitations = async (oracle) => {
+  let { agoricNames, fromBoard, vstorage } = await makeRpcUtils({ fetch });
+
+  let feedBoards = agoricNames.reverse;
+
+  let feedInvs = {};
+
+  const current = await getCurrent(String(oracle), fromBoard, { vstorage });
+  const invitations = current.offerToUsedInvitation;
+
+  // For each invitation
+  for (let inv in invitations) {
+    let boardId = invitations[inv].value[0].instance.boardId;
+    let feed = feedBoards[boardId].split(" price feed")[0];
+
+    feedInvs[feed] = Number(inv);
+  }
+
+  return feedInvs;
+};
+
+/**
+ * Function to query round from chain
+ * @param {string} feed feed name of the price to query (Ex. ATOM-USD)
+ * @returns {RoundDetails} the latest round
+ */
+export const queryRound = async (feed) => {
+  // Read value from vstorage
+  const capDataStr = await readVStorage(feed, true);
+
+  let capData;
+
+  try {
+    //parse the value
+    capData = JSON.parse(JSON.parse(capDataStr).value);
+    capData = JSON.parse(capData.values[capData.values.length - 1]);
+    // Replace any extra characters
+    capData = JSON.parse(capData.body.replaceAll("\\", ""));
+  } catch (err) {
+    throw new Error("Failed to parse CapData for queryRound");
+  }
+
+  // Get round from result
+  let round = Number(capData.roundId.digits);
+
+  // Get offers
+  let offers = await getOraclesInvitations(envvars.FROM);
+
+  // Check if invitation for feed exists
+  if (!(feed in offers)) {
+    throw new Error(
+      "Invitation for " + feed + " not found in oracle invitations"
+    );
+  }
+
+  // Get feed offer id
+  let feedOfferId = offers[feed];
+
+  // Check if there is a submission for round
+  let submissionForRound = await checkSubmissionForRound(
+    envvars.FROM,
+    feedOfferId,
+    round
+  );
+
+  // Get the latest round
+  let latestRound = new RoundDetails(
+    round,
+    Number(capData.startedAt.digits),
+    capData.startedBy,
+    submissionForRound
+  );
+
+  logger.info(feed + " Latest Round: " + latestRound.roundId);
+  return latestRound;
+};
+
+/** @param {import('../lib/psm.js').BridgeAction} bridgeAction */
+export const outputAction = (bridgeAction) => {
+  return marshaller.serialize(bridgeAction);
+};
+
+/**
+ * Function to push price on chain to the smart wallet
+ * @param {number} price price to push
+ * @param {string} feed feed to push price to (Ex. ATOM-USD)
+ * @param {number} round round to push result to
+ * @param {string} from account to push from
+ * @returns {boolean} whether successful
+ */
+export const pushPrice = async (price, feed, round, from) => {
+  // Create an offerId with the Date number
+  let offerId = Date.now();
```

Comment:
Not used

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/29244ced3c6d34785c4643507480179d98d7d8c6